File: Projects/CPP/src/Plants/Dandelion.cpp
#include "Dandelion.h"
#include "../World.h"

Dandelion::Dandelion(Position&& position) : Plant(0, "\033[38;5;11m*\033[m", std::move(position)) {}

const std::string Dandelion::Type = "Dandelion";

std::string Dandelion::GetType() const {
    return Dandelion::Type;
}

Organism* Dandelion::GetNewOfType(Position&& position) {
    return new Dandelion(std::move(position));
}

void Dandelion::HandleAction(World& world) {
    int originalNumberOfOrganisms = world.GetNumberOfLivingOrganisms();
    for (int i = 0; i < 3; ++i) {
        Plant::HandleAction(world);
        if (world.GetNumberOfLivingOrganisms() > originalNumberOfOrganisms) {
            break;
        }
    }
}

File: Projects/Java/src/main/java/com/nieradko/worldsim/core/plants/Dandelion.java
package com.nieradko.worldsim.core.plants;

import com.nieradko.worldsim.core.IActionContext;
import com.nieradko.worldsim.core.Position;
import javafx.scene.paint.Color;

public class Dandelion extends Plant {
    public Dandelion(Position position) {
        super(0, Color.YELLOW, position);
    }

    @Override
    protected void handleAction(IActionContext context) {
        var numberOfLivingOrganisms = context.getNumberOfLivingOrganisms();
        for (var i = 0; i < 3; i++) {
            super.handleAction(context);
            if (numberOfLivingOrganisms < context.getNumberOfLivingOrganisms()) {
                break;
            }
        }
    }
}
